### PR TITLE
Fixes #646 - Investigate failure of test_that_checks_the_deletion_of_a_review and test_that_checks_the_editing_of_a_review

### LINKED ISSUE #646: Investigate failure of test_that_checks_the_deletion_of_a_review and test_that_checks_the_editing_of_a_review
These have failed on `dev` [1],  `dev.saucelabs` [2], [3] and `stage.saucelabs` [4].

[1] https://webqa-ci.mozilla.com/job/marketplace.dev/91/testReport/tests.desktop.consumer_pages.test_reviews/TestReviews/test_that_checks_the_deletion_of_a_review/
[2] https://webqa-ci.mozilla.com/job/marketplace.dev.saucelabs/122/testReport/tests.desktop.consumer_pages.test_reviews/TestReviews/test_that_checks_the_editing_of_a_review/
[3] https://webqa-ci.mozilla.com/job/marketplace.dev.saucelabs/119/testReport/tests.desktop.consumer_pages.test_reviews/TestReviews/test_that_checks_the_deletion_of_a_review/
[4] https://webqa-ci.mozilla.com/job/marketplace.stage.saucelabs/131/testReport/tests.desktop.consumer_pages.test_reviews/TestReviews/test_that_checks_the_editing_of_a_review/

### PR DESCRIPTION
**Pull request by bobsilverberg**
Address issues causing tests to fail

Use a unique user per test and create a new review each time
Locate a review by a user by their username, rather than by the fact that there is a delete link
Rename ReviewSnippet to Review

This should stabilize these tests which have been having a lot of issues.

@m8ttyB / @rbillings r?

### CODE REVIEW

File: pages/desktop/consumer_pages/reviews.py [hunk 1]
```diff
@@ -27,24 +27,29 @@ def __init__(self, testsetup, app_name=False):
...
+        return [self.Review(self.testsetup, web_element)
                 for web_element in self.selenium.find_elements(*self._review_locator)]
 
-    @property
-    def logged_in_users_review(self):
+    def get_review_for_user(self, user_name):
         for review in self.reviews:
-            if review.find_element(*review._delete_review_locator):
+            if review.author == user_name:
                 break
```
Comment (m8ttyB) [inline pages/desktop/consumer_pages/reviews.py]: Honestly might be due to #toddlerlife #nosleep effecting me, should `return review` be moved into this `if` block?

File: tests/desktop/consumer_pages/test_reviews.py [hunk 2]
```diff
@@ -16,36 +14,20 @@
...
+    def _create_review(self, mozwebqa):
         # Step 1 - Login into Marketplace
         mock_review = MockReview()
         home_page = Home(mozwebqa)
         home_page.go_to_homepage()
 
         home_page.header.click_sign_in()
-        acct = self.create_new_user(mozwebqa)
-        home_page.login(acct)
+        self.acct = self.create_new_user(mozwebqa)
```
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_reviews.py]: Can you create a code level comment as to why the use of `self.`?

File: pages/desktop/consumer_pages/reviews.py (hunk 1 above)
Comment (bobsilverberg) [inline pages/desktop/consumer_pages/reviews.py]: You are absolutely right, @m8ttyB. This method could be greatly improved. I'll fix it tomorrow and ask for another review. Thanks!

File: tests/desktop/consumer_pages/test_reviews.py (hunk 2 above)
Comment (m8ttyB) [inline tests/desktop/consumer_pages/test_reviews.py]: Also, small question, why a class level `self.` here?

**Comment by bobsilverberg** [comment]
@m8ttyB I fixed up the `get_review_for_user` method and added a comment about why we are storing the `acct` variable in `self`.

Unfortunately these tests won't currently pass on dev due to the new nav which has broken pretty much all of our tests. I hope to get a PR out for that breakage soon.

**Comment by m8ttyB** [comment]
r+

**Comment by m8ttyB** [comment]
merging

**Pull request merged**